editoast/src/views/train_schedule/mod.rs

```diff
@@ -104,23 +109,74 @@ async fn delete(db_pool: Data<DbPool>, train_schedule_id: Path<i64>) -> Result<H
 async fn patch(
     db_pool: Data<DbPool>,
     train_schedules_changeset: Json<Vec<TrainScheduleChangeset>>,
+    core: Data<CoreClient>,
 ) -> Result<HttpResponse> {
     let train_schedules_changeset = train_schedules_changeset.into_inner();
-    let mut train_schedules = Vec::<TrainSchedule>::new();
-    for changeset in train_schedules_changeset {
-        let id = changeset.id.ok_or(TrainScheduleError::NoPatchId)?;
-        let train_schedule = match changeset.update(db_pool.clone(), id).await? {
-            Some(ts) => ts,
-            None => {
-                return Err(TrainScheduleError::NotFound {
-                    train_schedule_id: id,
+    let mut conn = db_pool.get()?;
+    conn.transaction::<_, InternalError, _>(|conn| {
+        let mut train_schedules = Vec::new();
+        for changeset in train_schedules_changeset {
+            let id = changeset.id.ok_or(TrainScheduleError::NoPatchId)?;
+            let train_schedule: TrainSchedule = match changeset.update_conn(conn, id)? {
+                Some(ts) => ts,
+                None => {
+                    return Err(TrainScheduleError::NotFound {
+                        train_schedule_id: id,
+                    }
+                    .into())
+                }
+            }
+            .into();
+
+            // Delete the associated simulation output
+            {
+                use crate::tables::osrd_infra_simulationoutput::dsl::*;
+                use diesel::prelude::*;
+                match osrd_infra_simulationoutput
+                    .filter(train_schedule_id.eq(train_schedule.id.unwrap()))
+                    .get_result::<SimulationOutput>(conn)
+                {
+                    Ok(simulation_output) => {
+                        SimulationOutput::delete_conn(conn, simulation_output.id)?;
+                    }
+                    Err(diesel::result::Error::NotFound) => {}
+                    Err(err) => return Err(err.into()),
                 }
-                .into())
             }
+
+            train_schedules.push(train_schedule);
         }
-        .into();
-        train_schedules.push(train_schedule);
-    }
+
+        // Resimulate the trains
+        let id_timetable = train_schedules[0].timetable_id;
```

Comment:
Shouldn't you check the length of `train_schedules` first ? So as to return a 400 if it's empty instead of a 500